Kernel/Process.cpp

```diff
@@ -1086,6 +1089,44 @@ int Process::do_exec(NonnullRefPtr<FileDescription> main_program_description, Ve
     return 0;
 }
 
+Vector<AuxiliaryValue> Process::generate_auxiliary_vector()
+{
+    Vector<AuxiliaryValue> auxv;
+    // PHDR/EXECFD
+    // PH*
+    auxv.append({ AuxiliaryValue::PageSize, PAGE_SIZE });
+    auxv.append({ AuxiliaryValue::BaseAddress, (void*)m_load_offset });
+    // FLAGS
+    auxv.append({ AuxiliaryValue::Entry, (void*)m_entry_eip });
+    // NOTELF
+    auxv.append({ AuxiliaryValue::Uid, (long)m_uid });
+    auxv.append({ AuxiliaryValue::EUid, (long)m_euid });
+    auxv.append({ AuxiliaryValue::Gid, (long)m_gid });
+    auxv.append({ AuxiliaryValue::EGid, (long)m_egid });
+
+    // FIXME: Don't hard code this? We might support other platforms later.. (e.g. x86_64)
+    auxv.append({ AuxiliaryValue::Platform, "i386" });
+    // FIXME: This is platform specific
+    auxv.append({ AuxiliaryValue::HwCap, (long)CPUID(1).edx() });
+
+    // FIXME: Is this 100HZ or 1000HZ?
+    auxv.append({ AuxiliaryValue::ClockTick, 100 });
+
+    // FIXME: Also take into account things like extended filesystem permissions? That's what linux does...
+    auxv.append({ AuxiliaryValue::Secure, ((m_uid != m_euid) || (m_gid != m_egid)) ? 1 : 0 });
+
+    // +1 for null term, we're going to be lazy and store this as a string
+    char random_bytes[17] {};
+    get_good_random_bytes((u8*)random_bytes, 16); // FIXME: Are these "secure random bytes?"
+
+    auxv.append({ AuxiliaryValue::Random, String(random_bytes) });
```

Comment:
The `String(const char*)` constructor will stop fetching bytes if it encounters a null terminator.
So if you get a zero early in the string of random bytes, we'll have a much shorter sequence as a result.

Reply:
Oh whoops. Was expecting it to select a `template <size_t> String(const char (&array)[N]) ` sort of constructor instead. Changed to char* + length constructor